internal/conversion.go

```diff
@@ -0,0 +1,78 @@
+package internal
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/hazelcast/hazelcast-go-client/serialization"
+)
+
+// supported types
+const (
+	String  = "string"
+	Boolean = "boolean"
+	JSON    = "json"
+	Int8    = "int8"
+	Int16   = "int16"
+	Int32   = "int32"
+	Int64   = "int64"
+	Float32 = "float32"
+	Float64 = "float64"
+)
+
+var SupportedTypes = []string{
+	String,
+	Boolean,
+	JSON,
+	Int8,
+	Int16,
+	Int32,
+	Int64,
+	Float32,
+	Float64,
+}
+
+func ConvertString(value, valueType string) (interface{}, error) {
+	var (
+		convertedValue interface{}
+		err            error
+	)
+	var (
+		intermediateInt   int64
+		intermediateFloat float64
```

Comment:
These variable names are too long. How about just `i` and `f`?

Reply:
I was going to name them tmp, but figured you would not be happy with that 😬